### ISSUE TITLE
Singleton connections 

### ISSUE THREAD

**Issue by ywadi** [SpectrumBroad/xible#93]
Hello, this is an awesome project! 
Just need to clarify something as it is not in the docs, or might not have been in the scope. 
I am creating a pub/sub connector to MQTT, I want to be able to connect to the server once with the username and password. But then share that connection with all the nodes that will be using the MQTT connection. Is that possible ? If not, is it in the scope? If it is I would be happy to help on that if you point out where this needs to be included

**Comment by ywadi** [comment]
Thanks for your response @steve2507 , 
This kind of works for each workflow, let me put it this way, if you have multiple workflows then we will have multiple connections to the mongodb database if each workflow has the mongodb connection node, which is not the best practice in this case for performance on both end (the db and xible) as this will have multiple open connections. 
I was assuming the ability to create 1 mongodb connection that can be shared between multiple workflows, so a global connection to all workflows where the connection object can be shared.
Hope that clarifies it?

**Comment by steve2507** [comment]
Yeah any thoughts on this would be appreciated. Back when I used Xible for home automation, I had a similar 'issue' and was thinking about creating a separate 'global flow' or something like that to store nodes which are exposed to every flow available.

**Comment by ywadi** [comment]
A suggestion would be that the node that has been created that will be "connecting to a server or serving a global connection" would contain an executable file defined in the structure and the execution of this code returns an object (server connection for example) as a singleton. EXAMPLE:
Mongodb connect to server node will contain the following; 
- Structure File : Add description somehow that it will contain connection defs. that are generated as an object form the editor "Many connection types that can be used on multiple nodes" which would contain as an example ("server host, port, username, password")
- Index.js File : <Not Changed Possibly> 
- SingletonDefinitions.js "Need a better name" : Which will contain the ability to load the configurations defined on the editor as an object, executed and saved as a singleton. 

when using this node, the user will have a dropdown list asking which configuration it would use (the predefined configurations). 

Might be tricky to explain in text for now but do you get the gist of it ?

**Comment by steve2507** [comment]
You already concluded this I see, but indeed every flow runs as its own process to not hinder each other in the case of computational intensive tasks. This will also allow for clustering a Xible setup later versions.

I do believe what you need is possible, without any extra development from a nodepack developer, simply by letting a Xible user define some nodes (such as the mongodb one) in a central place.

**Comment by ywadi** [comment]
Would you have a suggestion on how to proceed with that ? 
The multi-process approach makes total sense

**Comment by steve2507** [comment]
Yes, I think so.
I'm currently working on a solution where a flow can import any other flow (the flow definition, not a flow instance). This would make it possible to simply store connection nodes and such in one flow and include that flow in others.
A next step would be to make a special place available for a 'global' flow that by default is included in all flows.
